src/rgw/rgw_rest_sts.cc

```diff
@@ -271,28 +270,28 @@ WebTokenEngine::get_from_jwt(const DoutPrefixProvider* dpp, const std::string& t
       }
       if (! found && ! is_client_id_valid(provider.client_ids, client_id) && ! is_client_id_valid(provider.client_ids, azp)) {
         ldpp_dout(dpp, 0) << "Client id in token doesn't match with that registered with oidc provider" << dendl;
-        throw -EACCES;
+        throw std::system_error(EACCES, std::system_category());
       }
     }
     //Validate signature
     if (decoded.has_algorithm()) {
-      auto& algorithm = decoded.get_algorithm();
+      auto algorithm = decoded.get_algorithm();
       try {
         validate_signature(dpp, decoded, algorithm, iss, provider.thumbprints, y);
-      } catch (...) {
-        throw -EACCES;
+      } catch (const std::exception& e) {
+        throw std::system_error(EACCES, std::system_category());
       }
     } else {
       return {boost::none, boost::none};
     }
   } catch (int error) {
     if (error == -EACCES) {
-      throw -EACCES;
+      throw std::system_error(EACCES, std::system_category());
```

Comment:
do we still need a catch block for `int error`?

Reply:
probably can remove now, will check once